lib/chef/knife/vsphere_vm_nic_delete.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Author:: Scott Williams (<scott@backups.net.au>)
+# License:: Apache License, Version 2.0
+#
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+class Chef::Knife::VsphereVmNicDelete < Chef::Knife::BaseVsphereCommand
+  banner 'knife vsphere vm nic delete VMNAME NICNAME'
+
+  common_options
+
+  def run
+    $stdout.sync = true
+
+    vmname = @name_args[0]
+    if vmname.nil?
+      show_usage
+      fatal_exit('You must specify a virtual machine name')
+    end
+
+    nicname = @name_args[1]
+    if nicname.nil?
+      show_usage
+      fatal_exit('You must specify the name of the NIC to delete')
+    end
+
+    vim_connection
+    vm = get_vm(vmname) || abort('VM not found')
+
+    vm.config.hardware.device.each.grep(RbVmomi::VIM::VirtualEthernetCard) do |a|
```

Comment:
This might be more clear if you were to find the NIC and then delete it, rather than deleting it as part of the `each`. It would also let you throw an error if the NIC wasn't found. What about

```Ruby
card = vm.config.hardware.device.find { |c| c.is_a? RbVmomi::VIM::VirtualEthernetCard && c.deviceInfo.label == nicname }
if card.nil?
  fatal_exit "Could not find a nic called #{nicname}"
end
...
```

Or, closer to what you had, and a bit more friendly for the user:

```Ruby
cards = vm.config.hardware.device.grep(RbVmomi::VIM::VirtualEthernetCard)
card = cards.find { |c| c.deviceInfo.label == nicname }
if card.nil?
  found = cards.map { |c| c.deviceInfo.label}.join ', '
  fatal_exit "Could not find #{nicname}. I did find #{found}."
end
...
```